includes/managers/widgets.php

```diff
@@ -552,6 +552,69 @@ public function get_widgets_frontend_settings_keys() {
 		return $keys;
 	}
 
+	/**
+	 * Widgets styles names.
+	 *
+	 * Some Elementor widgets have custom styles. Each widget needs to register
+	 * the styles is depends on. Later, based on the loading strategy, Elementor
+	 * chooses how and where to enqueue these styles.
+	 *
+	 * Elementor has two types of widgets, widgets in the `/includes/widgets/`
+	 * folder and widgets in the `/modules/` folder.
+	 *
+	 * The styles are compiled to `/assets/css/widget-{$widget_name}.css` files
+	 * using grunt tasks.
+	 *
+	 * Widgets in the `/includes/widgets/` folder compile their SCSS files from
+	 * the `/assets/dev/scss/frontend/widgets/` folder.
+	 * Widgets in the `/modules/` folder compile their SCSS files from the
+	 * `/modules/{$module_name}/assets/scss/` folder.
+	 *
+	 * This method returns the list of all the widgets in the `/includes/`
+	 * folder that have styles.
+	 *
+	 * @since 3.24.0
+	 * @access public
+	 *
+	 * @return array The names of the widgets styles.
+	 */
+	public function widgets_styles_names(): array {
```

Comment:
I am curious, how can this be extended by other plugins, including the Pro plugin?

Reply:
This is a unique solution for internal widgets in the `/includes/widgets/` folder. Widgets in the `/modules/` folder don't need this, so are third parties.

Third party developers can use the docs:
https://developers.elementor.com/docs/widgets/widget-dependencies/